src/dmd/backend/backconfig.c

```diff
@@ -401,7 +420,7 @@ void util_set64()
     _tysize[TYnullptr] = 8;
     _tysize[TYnptr] = 8;
     _tysize[TYnref] = 8;
-#if TARGET_LINUX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_SOLARIS || TARGET_OSX
+#if TARGET_LINUX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_DRAGONFLYBSD || TARGET_SOLARIS || TARGET_OSX
```

Comment:
It is ridiculous that every UNIX has to be added to this list so many places.  There needs to be a `TARGET_POSIX` and maybe other defines, so that this common list can be maintained in one header.  Not saying you should do it, just saying it shouldn't be this much work to add a new UNIX like Dragonfly.

Reply:
Backend can do as they please, but at some point I hope to remove all traces of `TARGET_XXX` in the frontend except the target module, where the logic will now live. (`global.params.isXXX` will be removed and made private vars in target also).

Reply:
I do agree, especially when it comes to an OS that clearly comes from a particular family (BSD in this case). Initially I wanted to add a TARGET_BSD and have all BSD's inherit from that, which would simplify a lot of these cases.

I did contact a couple of people and checked other related PR's before going down this road, and this issue has been raised multiple/several times before. @WalterBright (amongst others) was strongly against mixing/reusing code and wants a separate definition for each architecture and os. 

This does have mean quite a bit of code duplication and added maintenance. But on the other hand, it does have a positive effect, namely changes made for DragonFly have 0 change of messing stuff up for FreeBSD, OpenBSD, Linux, Darwin etc. The latter was the main reason to implement it this way (If i am not mistaken).

Keeping this positive effect in place, there are still possibilities for refactoring. For example these large 'version(xxxx)' trees could be split into a single file per architecture and OS (You would still have code duplication but an easier job to port to a new OS. Namely: one single backend file to edit).

If you do see reason to re-open this discussion, i think it should take place elsewhere, and not this PR.

Reply:
>WalterBright (amongst others) was strongly against mixing/reusing code and wants a separate definition for each architecture and os.

I think you misunderstood.  Walter wants each OS to be versioned off separately for most D code, but clearly this compiler code is already not being separated, which is why you're just adding to this list.  I believe he'd be fine with merging all this into one define or eventually one D version, as Iain plans to do exactly that.

Anyway, I'm not reopening this discussion, as I noted I'm not asking you to consolidate, just making a comment that this needs to be changed, which Iain has acknowledged as the plan.

Reply:
I see what you mean now. Adding a couple of generic TARGET groups (POSIX/BSD/WIN/OSX) would be quite nice, and clear up some of these long repetitive ifdefs. Hopefully this refactorying will happen after this PR has already been merged :-)